lib/core-common/src/utils/to-importFn.ts

```diff
@@ -1,19 +1,21 @@
+import globBase from 'glob-base';
+import { makeRe } from 'micromatch';
 import dedent from 'ts-dedent';
 
 import type { NormalizedStoriesEntry } from '../types';
-import { toRequireContext } from './to-require-context';
 
 export function toImportFnPart(entry: NormalizedStoriesEntry) {
-  const { path: base, match } = toRequireContext(entry.glob);
+  const { base } = globBase(entry.glob);
+  const regex = makeRe(entry.glob, { fastpaths: false, noglobstar: false, bash: false });
```

Comment:
You know I think if we just exported `regex` from `toRequireContext` this would have worked. What I didn't check closely enough is that `toRequireContext` later messes with the RE.
```suggestion
  const { base, regex } = toRequireContext(entry.glob)
```

Reply:
Hi @tmeasday that seems to work. However, I don't know how to update all the logic in `toRequireContext` to handle the various cases. Is that something you could take a crack at?

Reply:
Done!